src/Exceptionless.Core/Jobs/EventPostsJob.cs

```diff
@@ -30,47 +30,77 @@ public class EventPostsJob : QueueJobBase<EventPost> {
         private readonly IOrganizationRepository _organizationRepository;
         private readonly IProjectRepository _projectRepository;
         private readonly IFileStorage _storage;
+        private readonly JsonSerializerSettings _jsonSerializerSettings;
 
-        public EventPostsJob(IQueue<EventPost> queue, EventParserPluginManager eventParserPluginManager, EventPipeline eventPipeline, IMetricsClient metricsClient, IOrganizationRepository organizationRepository, IProjectRepository projectRepository, IFileStorage storage, ILoggerFactory loggerFactory = null) : base(queue, loggerFactory) {
+        public EventPostsJob(IQueue<EventPost> queue, EventParserPluginManager eventParserPluginManager, EventPipeline eventPipeline, IMetricsClient metricsClient, IOrganizationRepository organizationRepository, IProjectRepository projectRepository, IFileStorage storage,
+            JsonSerializerSettings jsonSerializerSettings, ILoggerFactory loggerFactory = null) : base(queue, loggerFactory) {
             _eventParserPluginManager = eventParserPluginManager;
             _eventPipeline = eventPipeline;
             _metricsClient = metricsClient;
             _organizationRepository = organizationRepository;
             _projectRepository = projectRepository;
             _storage = storage;
+            _jsonSerializerSettings = jsonSerializerSettings;
 
             AutoComplete = false;
         }
 
         protected override async Task<JobResult> ProcessQueueEntryAsync(QueueEntryContext<EventPost> context) {
             var queueEntry = context.QueueEntry;
-            var eventPostInfo = await _storage.GetEventPostAndSetActiveAsync(queueEntry.Value.FilePath, _logger, context.CancellationToken).AnyContext();
-            if (eventPostInfo == null) {
+            var fileInfo = await _storage.GetFileInfoAsync(queueEntry.Value.FilePath).AnyContext();
+            if (fileInfo == null) {
+                await queueEntry.AbandonAsync().AnyContext();
+                return JobResult.FailedWithMessage($"Unable to retrieve post data info '{queueEntry.Value.FilePath}'.");
+            }
+
+            if (fileInfo.Size > Settings.Current.MaximumEventPostSize) {
```

Comment:
Yes, but it's probably to be safer than sorry, since we are getting massive events in storage. Also overage handler might not work if the content length isn't sent or is malformed??